builtin/game/chat.lua

```diff
@@ -1034,12 +1034,11 @@ core.register_chatcommand("time", {
 		end
 		local hour, minute = param:match("^(%d+):(%d+)$")
 		if not hour then
-			local new_time = tonumber(param)
-			if not new_time then
-				return false, S("Invalid time.")
+			local new_time = tonumber(param) or -1
+			if new_time ~= new_time or new_time < 0 or new_time > 24000 then
```

Comment:
```new_time ~= new_time``` is always false
can't that be left out?

Reply:
This is a NaN check

Reply:
Should probably be `core.is_nan` then

Reply:
I don't think `tonumber` ever returns NaN.

Reply:
Thanks. Lua keeps surprising me. That is so not intuitive.

Reply:
> Thanks. Lua keeps surprising me. That is so not intuitive.

That's not specific to Lua though - languages which comply to the float spec (IEEE 754) have to do it this way.